File: files/problems/cpp/count-the-number-of-consistent-strings.cpp
// by @codeAbinash

bool matchMap(bool *mp1, bool *mp2) {
    for (int i = 0; i < 26; i++)
        if (mp2[i] && !mp1[i]) return false;
    return true;
}

class Solution {
   public:
    int countConsistentStrings(string allowed, vector<string> &words) {
        bool mp[26] = {0};
        for (int i = 0; i < allowed.length(); i++) mp[allowed[i] - 'a'] = true;
        int count = 0, size = words.size();
        for (int i = 0; i < size; i++) {
            bool mpWord[26] = {0};
            int len = words[i].length();
            for (int j = 0; j < len; j++) mpWord[words[i][j] - 'a'] = true;
            if (matchMap(mp, mpWord)) count++;
        }
        return count;
    }
};

File: leetcode/problems/java/count-the-number-of-consistent-strings.java
class Solution {
  public int countConsistentStrings(String allowed, String[] words) {
    int arr[] = new int[26];
    for (char i : allowed.toCharArray())
      arr[i - 'a'] = 1;
    int count = 0;
    for (String k : words)
      count += findConsistentString(arr, k);
    return count;
  }

  public static int findConsistentString(int arr[], String k) {
    int flag = 1;
    for (int i = 0; i < k.length(); i++) {
      if (arr[k.charAt(i) - 'a'] == 0) {
        flag = 0;
        break;
      }
    }
    return flag;
  }
}